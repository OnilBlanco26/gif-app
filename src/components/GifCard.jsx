import PropTypes from 'prop-types'

const GifCard = ({title, url}) => {



  return (
    <div className='rounded-md w-auto h-auto overflow-hidden bg-white text-black flex flex-col items-center flex-wrap shadow-md shadow-black'>
        <img className='w-full h-auto' src={url} alt={title} />
        <h3 className=" text-lg text-center font-bold uppercase mx-4">{title}</h3>
        {/* <a href={url} download={true} target='_blank'>
            <button>Download</button>
        </a> */}
    </div>
  )
}

GifCard.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}



export default GifCard