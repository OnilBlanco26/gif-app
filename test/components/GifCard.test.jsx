import { render, screen } from "@testing-library/react"
import GifCard from "../../src/components/GifCard"
/* eslint-disable no-undef */

describe('Pruebas en <GifItem />', () => {

    const title = 'Onil'
    const url = 'https://prueba/onil/search.jpg'


    test('Debe de hacer match con el snapshot', () => {

       const {container} = render(<GifCard title={title} url={url} />)

       expect(container).toMatchSnapshot()
       
    })


    test('debe de mostrar la imagen con el URL  y el ALT indicado', () => {
        // const {container} = render(<GifCard title={title} url={url} />)
        // screen.debug()

        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)

        const {src,alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(alt)
    })

    test('Debe de mostrar el titulo en en componente', () => {

        render( <GifCard title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy()
        
    })
    
    
  
  
})




/* eslint-enable no-undef */