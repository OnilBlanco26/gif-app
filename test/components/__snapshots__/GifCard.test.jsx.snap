// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="rounded-md w-auto h-auto overflow-hidden bg-white text-black flex flex-col items-center flex-wrap shadow-md shadow-black"
  >
    <img
      alt="Onil"
      class="w-full h-auto"
      src="https://prueba/onil/search.jpg"
    />
    <h3
      class=" text-lg text-center font-bold uppercase mx-4"
    >
      Onil
    </h3>
  </div>
</div>
`;
